smarts/core/traffic_history.py

```diff
@@ -230,6 +285,85 @@ def vehicles_active_between(
         rows = self._query_list(query, (start_time, end_time))
         return (TrafficHistory.VehicleRow(*row) for row in rows)
 
+    def vehicle_slice_in_range(
+        self, exists_at_or_after, ends_before, minimum_history_duration
+    ):
+        """Find all vehicles active between the given history times."""
+        query = """SELECT V.id, V.type, V.length, V.width, V.height,
+                          S.position_x, S.position_y, S.heading_rad, S.speed, D.avg_speed,
+                          S.sim_time, E.sim_time,
+                          E.position_x, E.position_y, E.heading_rad
+                   FROM Vehicle AS V
+                   INNER JOIN ( 
+                    SELECT vehicle_id, AVG(speed) as "avg_speed"
+                    FROM (SELECT vehicle_id, sim_time, speed from Trajectory WHERE sim_time >= ? AND sim_time < ?)
+                    GROUP BY vehicle_id
+                   ) AS D ON V.id = D.vehicle_id
+                   JOIN (
+                    SELECT
+                        vehicle_id,
+                        sim_time,
+                        speed,
+                        position_x,
+                        position_y,
+                        heading_rad
+                    FROM (SELECT * from Trajectory WHERE sim_time < ? ORDER BY sim_time ASC)
```

Comment:
Order this inner trajectory by simulation time so that it gets addressed in that order when next processed.